crates/iota-data-ingestion-core/src/worker_pool.rs

```diff
@@ -195,23 +212,48 @@ impl<W: Worker + 'static> WorkerPool<W> {
                             info!("received checkpoint for processing {} for workflow {}", sequence_number, task_name);
                             let start_time = Instant::now();
                             let backoff = backoff::ExponentialBackoff::default();
-                            backoff::future::retry(backoff, || async {
-                                worker
-                                    .clone()
+                            let result = backoff::future::retry(backoff, || async {
+                               let processed_checkpoint_result = worker
                                     .process_checkpoint(&checkpoint)
                                     .await
                                     .map_err(|err| {
                                         let err = IngestionError::CheckpointProcessing(err.to_string());
-                                        info!("transient worker execution error {:?} for checkpoint {}", err, sequence_number);
+                                        info!("transient worker execution error {err:?} for checkpoint {sequence_number}");
                                         backoff::Error::transient(err)
-                                    })
+                                    }).map(BackoffStatus::ProcessedCheckpoint);
+
+                                if processed_checkpoint_result.is_err() & token.is_cancelled() {
+                                    return Ok(BackoffStatus::Shutdown)
+                                }
+                                processed_checkpoint_result
```

Comment:
Couldn't we just use `WorkerStatus` instead here?

Reply:
That's a good observation, thank you. [Fixed](https://github.com/iotaledger/iota/pull/5496/commits/cf1ee82d3a6a4b38c5ab154824dd67d9df68ce7d)